torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -72,8 +72,13 @@ def check_node_safe(node: Node):
     Checks that the node only uses supported operators. We are starting with very
     conservative cacheability constraints, and incrementally adding more support as we expand.
     """
+    SAFE_TORCH_MODULES = ("torch.functional", "torch.nn.functional", "torch")
```

Comment:
What's the definition of "safe"? That there are no global side effects?

Reply:
There's a lot of private `torch._*` APIs that I'm a bit nervous about. If you want to be conservative, then only cache on the public APIs

Reply:
Are you going to also enforce some serializability constraint on the arguments to these functions?

Reply:
One category of functions to be aware of are things that save/load from files -- the contents could change from run-to-run. I would also hope that Dynamo already graph breaks on those. Another conservative thing to do here is eliminate functions with string constant arguments

Reply:
> One category of functions to be aware of are things that save/load from files -- the contents could change from run-to-run. I would also hope that Dynamo already graph breaks on those

Do you know of any examples of this? I guess `torch.load/save`, although I would hope that dynamo graph breaks on this for the reason you said (dynamo would have to guard on the tensor metadata of what it loaded, so the output of the load would need to be a graph input).

Although in terms of being conservative or not: I guess one question is to what extent the caching logic should be free to assume dynamo guard correctness when deciding what is safe to cache or not.

>  Another conservative thing to do here is eliminate functions with string constant arguments

The cache key is computed by serializing the FX graph and converting it into a key. So any hard-coded strings and ints that are inputs to nodes will end up in the cache key too?

Reply:
> The cache key is computed by serializing the FX graph and converting it into a key. So any hard-coded strings and ints that are inputs to nodes will end up in the cache key too?

Let's assume torch.load("foo.pt") is in the FX graph (it really shouldn't be, but for the sake of demonstration). Then, this node is "unsafe" because the contents of foo.pt can change between runs.

What I meant by "eliminating functions with string constant args" is to catch cases like this -- "foo.pt" can stay the same but it's referring to some location that can change.

Reply:
Sounds good - if we want to be overly conservative then we can definitely start by not including nodes that take in string arguments when deciding what is cache safe.

Although in the limit, we'll need to either:
* undo that and decide that we trust dynamo enough to include these in the cache
* build up an allow-list of "safe" ops that accept string inputs (e.g. `aten.div.rounding_mode`)